AWS cross account (#1717)
logzio/logz-docs: _source/logzio_collections/_log-sources/aws-cross-account.md

  - nshishkin
shipping-tags:
  - aws
order: 110
---
<!-- tabContainer:start -->
<div class="branching-container">

* [Overview](#overview)
* [Deploy integration](#deploy)
* [Update deployed integration](#update)
{:.branching-tabs}

<!-- tab:start -->
<div id="overview">

Deploy this integration to simultaneously ship logs from multiple AWS accounts to Logz.io.

The integration consists of the following main steps:

* Creating a **landing account**, which receives logs from your multiple AWS accounts and sends them to Logz.io.
* Creating **destinations in the landing account** on each region you need to send logs from.
* Creating **subscription filters** in your multiple AWS accounts to enable them to send logs from Cloudwatch to the **destinations in the landing account**.

</div>
<!-- tab:end -->

<!-- tab:start -->
<div id="deploy">

**Before you begin, you'll need**:

* Your AWS service [publishes logs to Cloudwatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/aws-services-sending-logs.html). 
* Your log group is in the format: `/<<AWS-PARTITION/<<SERVICE-NAME>>/<<LOG-GROUP-NAME>>`, for example: `/aws/lambda/my_function`.
* AWS CLI

<div class="tasklist">

> 

Reviewer: Can we add a sub-header to make it clear that steps 1-10 are in the landing account?

##### Specify the stack template

Keep the default settings in the **Create stack** screen and select **Next**.

##### Specify the stack details

Specify the stack details as per the table below and select **Next**.

| Parameter                | Description                                                                                                                                                                                                                                                                                             | Required / Defaults |
|--------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------|

> | `AccountsArns`           | Comma-delimited list (**no spaces**) of all ARNs involved in this integration. This includes ARNs of the landing account and the accounts you'd like to send logs from. The ARNs should be specified as follows: `arn:aws:logs:*:<<ACCOUNT_NUMBER>>:*,arn:aws:logs:*:<<ACCOUNT_NUMBER>>:*,arn:aws:logs:*:<<ACCOUNT_NUMBER>>:*`     | Required            |

Reviewer: ```suggestion
| `AccountsArns`           | Comma-delimited list (**no spaces**) of **all ARNs involved in this integration**. This includes ARNs of the **landing account** and the **accounts you'd like to send logs from**. The ARNs should be specified as follows: `arn:aws:logs:*:<<ACCOUNT_NUMBER>>:*`. For example, if our landing account is 123, and we also want to send logs from account 456, we'll insert: `arn:aws:logs:*:123:*,arn:aws:logs:*:456:*`    | Required            |
```

| `LogzioCOMPRESS`         | If true, the Lambda will send compressed logs. If false, the Lambda will send uncompressed logs.                                                                                                                                                                                                        | Default: `true`     |
| `LogzioREGION`           | Two-letter region code, or blank for US East (Northern Virginia). This determines your listener URL (where you're shipping the logs to) and API URL. You can find your region code in the Regions and URLs [here](https://docs.logz.io/user-guide/accounts/account-region.html#regions-and-urls table). | Default: `us`       |
| `LogzioTOKEN`            | The token of the account you want to ship to. Can be found [here](https://app.logz.io/#/dashboard/settings/general).                                                                                                                                                                                    | Required            |

> | `SendingAccounts`        | Comma-delimited list (**no spaces**) of account numbers of the accounts you'd like to send logs froms. The numbers should be specified as follows: `"1234","5678","9012"`                                                                                                                                                                  | Required           |

Reviewer: ```suggestion
| `SendingAccounts`        | Comma-delimited list (**no spaces**) of account numbers of the accounts you'd like to send logs from. **Each account number should be wrapped in double-quotes.** The numbers should be specified as follows: `"1234","5678","9012"`                                                                                                                                                                  | Required           |
```

##### Configure stack options

If required, specify the **Key** and **Value** parameters for the **Tags** and select **Next**.

##### Review the stack

Confirm that you acknowledge that AWS CloudFormation might create IAM resources, IAM resources with custom names, may require `CAPABILITY_AUTO_EXPAND` and select **Create stack**.

> 

Reviewer: Can we add a note here that explains that steps 6-10 are only required if they want to send logs from regions that are not the region they deployed their main stack in?
Because the main stack already creates a destination for the region they deployed the main stack in.

##### Create a stack to deploy destinations in the landing account on each region you need to send logs from

Set the region of your AWS account to `us-east-1` and launch the stach by clicking the button below:

> [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://integrations-testing.s3.amazonaws.com/cross-accounts/0.0.1/sam-template-destination.yaml)

Reviewer: We will have this stack available at every region, so it won't have to be in `us-east-1`. I'll give you a complete table with regions and buttons once i'll deploy them.
It will be something like that:
| AWS Region | Deployment |
| --- | --- |
| `us-east-1` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://integrations-testing.s3.amazonaws.com/cross-accounts/0.0.1/sam-template-main.yaml) |

##### Specify the stack template

Keep the default settings in the **Create stack** screen and select **Next**.

##### Specify the stack details

Specify the stack details as per the table below and select **Next**.

| Parameter          | Description                                                                                                                                                 |
|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `RoleArn`          | The ARN of the **Role** created in the main stack. You can find it in the main stack's **Outputs** tab, under **LogzioCrossAccountRole**                        |
| `KinesisStreamArn` | The ARN of the **Kinesis Stream** created in the main stack. You can find it in the main stack's **Outputs** tab, under **LogzioCrossAccountKinesisStreamArn**. |

> | `SendingAccounts`        | Comma-delimited list (**no spaces**) of account numbers of the accounts you'd like to send logs froms. The numbers should be specified as follows: `"1234","5678","9012"` | 

Reviewer: ```suggestion
| `SendingAccounts`        | Comma-delimited list (**no spaces**) of account numbers of the accounts you'd like to send logs from. **Each account number should be wrapped in double-quotes.**  The numbers should be specified as follows: `"1234","5678","9012"` | 
```

##### Configure stack options

If required, specify the **Key** and **Value** parameters for the **Tags** and select **Next**.

##### Review the stack

Confirm that you acknowledge that AWS CloudFormation might create IAM resources, IAM resources with custom names, may require `CAPABILITY_AUTO_EXPAND` and select **Create stack**.

> 

Reviewer: Can we add a sub-header so that it will be clearer that step 11 is in the sending accounts?

##### Create a subscription filter in your sending accounts

<!-- info-box-start:info -->
You need to create a subscription filter in each sending account separately.
{:.info-box.note}
<!-- info-box-end -->

1. Make sure your AWS CLI is connected to the account you want to send logs from.
2. Make sure you have set the CLI to the region of the account that you need to send logs from.
3. Run the following command:

```shell
aws logs put-subscription-filter \
    --log-group-name "<<LOG-GROUP-NAME>>" \
    --filter-name "<<SUBSCRIPTION-FILTER-NAME>>" \
    --filter-pattern " " \
    --destination-arn "<<DESTINATION-ARN>>"
```
   * Replace `<<LOG-GROUP-NAME>>` with the name of the log group you want to collect logs from.
   * Replace `<<SUBSCRIPTION-FILTER-NAME>>` with the name of the subscription filter you create.

>    * Replace `<<DESTINATION-ARN>>` with the ARN of the destination you created in the landing account. You can find the ARN in the main stack's **Outputs** tab. The destination will be in the same region as your log group. 

Reviewer: Not sure how to phrase it, but the Destination Arn doesn't have to be the one from the main stack, but it should be the arn of the destination that matches the region that they want to ship their logs from. For example - if the log stream is in `us-west-2`, then they should use the arn of the Destination that's in `us-west-2`.

> 

Reviewer: We also have an option to deploy it with Terraform:
https://github.com/logzio/aws-cross-account/blob/initial/README.md#create-a-subscription-filter-with-terraform

##### Check Logz.io for your logs

Give your logs some time to get from your system to ours, and then open [Kibana](https://app.logz.io/#/dashboard/kibana).

If you still don't see your logs, see [log shipping troubleshooting]({{site.baseurl}}/user-guide/log-shipping/log-shipping-troubleshooting.html).

</div>

</div>
<!-- tab:end -->

<!-- tab:start -->
<div id="update">

#### Update the integration

> 

Reviewer: Can we add some sort of an intro, that says that this is what they should do in case they already deployed the integration, and want to add more accounts to send logs from?

* Your log group is in the format: `/<<AWS-PARTITION/<<SERVICE-NAME>>/<<LOG-GROUP-NAME>>`, for example: `/aws/lambda/my_function`.
* AWS CLI

<div class="tasklist">

##### Select the main stack of the landing account

In your AWS Console, go to **Cloudformation**, choose your main stack and select **Update**.

##### Specify the stack template

Choose **Use current template** and select **Next**.

##### Specify the stack details

* Add new ARNs to `AccountsArns`. The ARNs should be specified as follows: `arn:aws:logs:*:<<ACCOUNT_NUMBER>>:*,arn:aws:logs:*:<<ACCOUNT_NUMBER>>:*,arn:aws:logs:*:<<ACCOUNT_NUMBER>>:*`.
* Add new account numbers to `SendingAccounts`. The numbers should be specified as follows: `"1234","5678","9012"`.

<!-- info-box-start:info -->
Do not overwrite existing values.
{:.info-box.note}
<!-- info-box-end -->

##### Configure stack options

If required, specify the **Key** and **Value** parameters for the **Tags** and select **Next**.

##### Review the stack

Confirm that you acknowledge that AWS CloudFormation might create IAM resources, IAM resources with custom names, may require `CAPABILITY_AUTO_EXPAND` and select **Create stack**.

> ##### Select the stack to deploy destinations in the landing account on each region you need to send logs from

Reviewer: ```suggestion
##### In each Destination Stack you deployed
```

  
* Landing account, which receives logs from your multiple AWS accounts and sends them to Logz.io.
* Sending accounts, which send logs from your AWS services to the landing account.
  
![Overview](https://dytvr9ot2sszz.cloudfront.net/logz-docs/aws-cross-account/overview-aws.png)  

The integration creates the following resources:
  
##### In the landing account

* Kinesis stream, which receives logs from multiple AWS accounts.
* Destination, which encapsulates the stream and allows to send the logs to it.
* Lambda function, which uses the Kinesis stream as a trigger, and sends the logs to Logz.io
* Additional Destinations (if required) for each region you need to ship logs from.
  
##### In the sending accounts

* Subscription filters, which send the logs from Cloudwatch to the Destination of the landing account.

</div>
<!-- tab:end -->

<!-- tab:start -->
<div id="deploy-aws">

**Before you begin, you'll need**:

* Your AWS service [publishes logs to Cloudwatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/aws-services-sending-logs.html). 

> * Your log group is in the format: `/<<AWS-PARTITION/<<SERVICE-NAME>>/<<LOG-GROUP-NAME>>`, for example: `/aws/lambda/my_function`.

Reviewer: ```suggestion
* Your log group is in the format: `/<<AWS-PARTITION>>/<<SERVICE-NAME>>/<<NAME>>`, for example: `/aws/lambda/my_function`.
```

* AWS CLI

<div class="tasklist">

#### Configure the landing account

##### Create a main stack to deploy the landing account

> Set the region of your AWS account to `us-east-1` and launch the stach by clicking the button below:
> 
> [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://integrations-testing.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)

Reviewer: ```suggestion
Click the button that matches the region you'd like to deploy your main stack to:

| AWS Region       | Launch a stack                                                                                                                                                                                                                                                                                                                                 |
|------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `us-east-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-us-east-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |
| `us-east-2`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-2#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-us-east-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |
| `us-west-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-west-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-us-west-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |
| `us-west-2`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-us-west-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |
| `eu-central-1`   | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-central-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-eu-central-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)     |
| `eu-north-1`     | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-north-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-eu-north-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)         |
| `eu-west-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-eu-west-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |
| `eu-west-2`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-west-2#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-eu-west-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |
| `eu-west-3`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-west-3#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-eu-west-3.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |
| `ca-central-1`   | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ca-central-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-ca-central-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)     |
| `ap-northeast-1` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-northeast-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-ap-northeast-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml) |
| `ap-northeast-2` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-northeast-2#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-ap-northeast-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml) |
| `ap-northeast-3` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-northeast-3#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-ap-northeast-3.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml) |
| `ap-south-1`     | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-south-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-ap-south-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)         |
| `ap-southeast-1` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-southeast-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-ap-southeast-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml) |
| `ap-southeast-2` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-southeast-2#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-ap-southeast-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml) |
| `sa-east-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=sa-east-1#/stacks/new?stackName=aws-cross-account-main&templateURL=https://logzio-aws-integrations-sa-east-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-main.yaml)           |

```

| `SendingAccounts`        | Comma-delimited list (**no spaces**) of account numbers of the accounts you'd like to send logs from. **Each account number should be wrapped in double-quotes.** The numbers should be specified as follows: `"1234","5678","9012"`                                                                                                                                                                  | Required           |

##### Configure stack options

If required, specify the **Key** and **Value** parameters for the **Tags** and select **Next**.

##### Review the stack

Confirm that you acknowledge that AWS CloudFormation might create IAM resources, IAM resources with custom names, may require `CAPABILITY_AUTO_EXPAND` and select **Create stack**.

##### Create a stack to deploy destinations in the landing account on each region you need to send logs from
  
<!-- info-box-start:info -->
This procedure is only required if you need to send logs from regions that are different to the region that the main stack is deployed in.
{:.info-box.note}
<!-- info-box-end -->
  

> Set the region of your AWS account to `us-east-1` and launch the stach by clicking the button below:
> 
> [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://integrations-testing.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)

Reviewer: ```suggestion

Click the button that matches the region you'd like to deploy your destination stack to:

| AWS Region       | Launch a stack                                                                                                                                                                                                                                                                                                                                               |
|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `us-east-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-us-east-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
| `us-east-2`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-2#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-us-east-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
| `us-west-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-west-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-us-west-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
| `us-west-2`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-us-west-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
| `eu-central-1`   | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-central-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-eu-central-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)     |
| `eu-north-1`     | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-north-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-eu-north-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)         |
| `eu-west-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-eu-west-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
| `eu-west-2`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-west-2#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-eu-west-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
| `eu-west-3`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=eu-west-3#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-eu-west-3.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
| `ca-central-1`   | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ca-central-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-ca-central-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)     |
| `ap-northeast-1` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-northeast-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-ap-northeast-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml) |
| `ap-northeast-2` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-northeast-2#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-ap-northeast-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml) |
| `ap-northeast-3` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-northeast-3#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-ap-northeast-3.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml) |
| `ap-south-1`     | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-south-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-ap-south-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)         |
| `ap-southeast-1` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-southeast-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-ap-southeast-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml) |
| `ap-southeast-2` | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=ap-southeast-2#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-ap-southeast-2.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml) |
| `sa-east-1`      | [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=sa-east-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://logzio-aws-integrations-sa-east-1.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)           |
```

  filter_pattern  = " "
  destination_arn = "<<DESTINATION-ARN>>"
}
```

   * Replace `<<LOG-GROUP-NAME>>` with the name of the log group you want to collect logs from.
   * Replace `<<SUBSCRIPTION-FILTER-NAME>>` with the name of the subscription filter you create.
   * Replace `<<DESTINATION-ARN>>` with the ARN of the destination that matches the region of the sending account that you want to ship logs from. For example, if the log stream is in `us-west-2`, then they should use the arn of the Destination that's in `us-west-2`. You can find the ARN in the main stack's **Outputs** tab.

<!-- info-box-start:info -->
If you create the log group and the subscription filter at the same time, add a `depends_on` field to the subscription filter and make it dependent on the log group, so that the log group will be created first.
{:.info-box.note}
<!-- info-box-end -->
  
##### Check Logz.io for your logs

Give your logs some time to get from your system to ours, and then open [Kibana](https://app.logz.io/#/dashboard/kibana).

If you still don't see your logs, see [log shipping troubleshooting]({{site.baseurl}}/user-guide/log-shipping/log-shipping-troubleshooting.html).

> 

Reviewer: **Note:** If your log group is not under the required format, the logs will arrive under the default type `aws-cross-account`. Otherwise, the type will be the service you sent the logs from.

</div>

</div>
<!-- tab:end -->

<!-- tab:start -->
<div id="update">

#### Update the integration
  
If you need to add more accounts to an existing integration, you can update the deployed stacks as follows. 

**Before you begin, you'll need**:

* Your AWS service [publishes logs to Cloudwatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/aws-services-sending-logs.html). 

> * Your log group is in the format: `/<<AWS-PARTITION/<<SERVICE-NAME>>/<<LOG-GROUP-NAME>>`, for example: `/aws/lambda/my_function`.

Reviewer: ```suggestion
* Your log group is in the format: `/<<AWS-PARTITION>>/<<SERVICE-NAME>>/<<NAME>>`, for example: `/aws/lambda/my_function`.
```

##### Configure stack options

If required, specify the **Key** and **Value** parameters for the **Tags** and select **Next**.

##### Review the stack

Confirm that you acknowledge that AWS CloudFormation might create IAM resources, IAM resources with custom names, may require `CAPABILITY_AUTO_EXPAND` and select **Create stack**.

##### Create a stack to deploy destinations in the landing account on each region you need to send logs from
  
<!-- info-box-start:info -->
This procedure is only required if you need to send logs from regions that are different to the region that the main stack is deployed in.
{:.info-box.note}
<!-- info-box-end -->
  

Set the region of your AWS account to `us-east-1` and launch the stach by clicking the button below:

> [![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://integrations-testing.s3.amazonaws.com/cross-accounts/0.0.1/sam-template-destination.yaml)

Reviewer: ```suggestion
[![Deploy to AWS](https://dytvr9ot2sszz.cloudfront.net/logz-docs/lights/LightS-button.png)](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new?stackName=aws-cross-account-destination&templateURL=https://integrations-testing.s3.amazonaws.com/aws-cross-accounts/0.0.1/sam-template-destination.yaml)
```
